src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,101 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
+
     private static final Injector injector = new Injector();
 
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        try {
+            return findFieldsCheckAnnotationsSetAccessibility(interfaceClazz);
+        } catch (IllegalAccessException e) {
+            throw new RuntimeException("Can't get instance of input class!", e);
+        }
+    }
+
+    private Object findFieldsCheckAnnotationsSetAccessibility(Class<?> interfaceClazz)
+            throws IllegalAccessException {
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field : declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = findFieldsCheckAnnotationsSetAccessibility(
+                        field.getType());
```

Comment:
There is a potential for infinite recursion here if there is a circular dependency between classes. Ensure that your design avoids such scenarios, or implement a mechanism to detect and handle them.